File: src/main/scala/com/github/mauricio/postgresql/parsers/ParameterStatusParser.scala
package com.github.mauricio.postgresql.parsers

import org.jboss.netty.buffer.ChannelBuffer
import com.github.mauricio.postgresql.ChannelUtils
import com.github.mauricio.postgresql.messages.backend.{ParameterStatusMessage, Message}
import java.nio.charset.Charset

/**
 * User: Maurício Linhares
 * Date: 2/25/12
 * Time: 7:06 PM
 */

class ParameterStatusParser( charset : Charset ) extends Decoder {

  import ChannelUtils._

  override def parseMessage(b: ChannelBuffer): Message = {
    new ParameterStatusMessage( readCString(b, charset), readCString(b, charset) )
  }

}

File: postgresql-async/src/main/java/com/github/mauricio/async/db/postgresql/parsers/ParameterStatusParser.kt
package com.github.mauricio.async.db.postgresql.parsers

import com.github.jasync.sql.db.util.ByteBufferUtils.readCString
import com.github.mauricio.async.db.postgresql.messages.backend.ParameterStatusMessage
import com.github.mauricio.async.db.postgresql.messages.backend.ServerMessage
import io.netty.buffer.ByteBuf
import java.nio.charset.Charset


class ParameterStatusParser(val charset: Charset) : MessageParser {

  override fun parseMessage(b: ByteBuf): ServerMessage {
    return ParameterStatusMessage(readCString(b, charset), readCString(b, charset))
  }

}